packages/opentelemetry-plugin-http/src/http.ts

```diff
@@ -404,7 +404,8 @@ export class HttpPlugin extends BasePlugin<Http> {
 
       const { origin, pathname, method, optionsParsed } = utils.getRequestInfo(
         options,
-        typeof args[0] === 'object' && typeof options === 'string'
+        typeof args[0] === 'object' &&
+          (typeof options === 'string' || options instanceof url.URL)
           ? (args.shift() as RequestOptions)
           : undefined
```

Comment:
perhaps we could change this part

```js
const { origin, pathname, method, optionsParsed } = utils.getRequestInfo(
        options,
        typeof args[0] === 'object' &&
          (typeof options === 'string' || options instanceof url.URL)
          ? (args.shift() as RequestOptions)
          : undefined);
```

by something like:

```js
const secondParamRequest = utils.getSecondParamRequest(....);
const { origin, pathname, method, optionsParsed } = utils.getRequestInfo(
        options, secondParamRequest);
```
WDYT ?

Reply:
I don't think we should do that because the `args` array is mutated which would be hard to understand if it happens in another file.

Reply:
what about something like:
```js
 const requestOptions = typeof args[0] === 'object' &&
          (typeof options === 'string' || options instanceof url.URL)
          ? (args.shift() as RequestOptions)
          : undefined;
const { origin, pathname, method, optionsParsed } = utils.getRequestInfo(
        options,
       requestOptions);
```
I just would like to make it clearer.
WDYT ?

Reply:
done